### PR TITLE
fix(agw): adding sqlite3 dependencies

### PR DESCRIPTION
**Pull request by vroon2703**
Signed-off-by: vroon2703 <varun27@fb.com>

## Summary
When cloud subscriber authentication is done, the MME writes to the subscriberdb SQLite database. This requires cpp sqlite functions. This PR adds the SQLite dependencies for those functions. 

## Test Plan
Before this change:
ls /usr/include/*sql* will return nothing
After this change:
ls /usr/include/*sql* will return 
/usr/include/sqlite3.h  /usr/include/sqlite3ext.h

### CODE REVIEW

File: lte/gateway/deploy/roles/magma/tasks/main.yml
Comment (nstng) [inline lte/gateway/deploy/roles/magma/tasks/main.yml:527]: Would this be better suited under "Install common Magma dependencies"?

File: lte/gateway/c/core/BUILD.bazel
Comment (nstng) [inline lte/gateway/c/core/BUILD.bazel]: Changes in this file must be done in #13470 - all files must be present during build time.

File: bazel/external/system_libraries.BUILD
Comment (nstng) [inline bazel/external/system_libraries.BUILD]: This change should be done in #13470.

File: lte/gateway/docker/services/c/Dockerfile
Comment (nstng) [inline lte/gateway/docker/services/c/Dockerfile:92]: For Bazel CI the dependency needs to be added to `.devcontainer/bazel-base/Dockerfile`. From a fast glance, I think mme is actually not build with Bazel in this Dockerfile - but imo it does not hurt to already add it (at some point mme will be build here with Bazel too).

**Review by nstng — CHANGES_REQUESTED** [review]
Hi, see comments for needed changes. You can test these changes in context with #13470 by cherry-picking this change on your local #13470 branch, rebuilding the VM or the bazel base container and running `bazel build ...` in magma root (be aware, the build will take some time because you do not have caches on your fresh setup).

FYI: the bazel CI runs on the bazel-base Docker container. This container is rebuild only after changes are merged back to master, i.e., CI on this PR will run on container without the sqlite library.

File: lte/gateway/c/core/BUILD.bazel
Comment (nstng) [inline lte/gateway/c/core/BUILD.bazel]: The test is failing because this library does not exists. This change also must be done in #13470.

**Review by nstng — CHANGES_REQUESTED** [review]

File: lte/gateway/deploy/roles/magma/tasks/main.yml
Comment (ssanadhya) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:527]: @nstng , @vroon2703  note that "Install common Magma dependencies" is a preburn action, so moving this library to that action will only be effective if we update the `magma-dev` box base image. It will not install the library if we just destroy and re-build the VM locally or in CI.

Comment (ssanadhya) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:527]: Since updating the magma-dev box image is out of scope for @vroon2703 's project plan, my suggestion would be to keep the install step for sqlite3 package as a `full_provision` action for this PR and someone from DevOps team can take over the box update task separately.

Comment (nstng) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:527]: @ssanadhya nice catch, I overlooked the "when" - sorry @vroon2703 you need your change from before with the comment that this dependency should be added to "Install common Magma dependencies" when the next preburn imaged is created.

**Review by ssanadhya — APPROVED** [review]
Lgtm

File: lte/gateway/deploy/roles/magma/tasks/main.yml
Comment (vroon2703) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:527]: Done!

**Comment by ssanadhya** [comment]
@nstng , I am force merging this to unblock PR #13470 . I checked that all the requested changes have been made and all required CI checks are passing. If further fixes are needed, we can create a new PR.

**Pull request merged**
